packages/svelte/src/internal/client/runtime.js

```diff
@@ -1004,6 +1037,21 @@ export function get(signal) {
  * @returns {V}
  */
 export function set(signal, value) {
+	if (DEV && 'owners' in signal) {
+		const component = get_component();
+
+		// @ts-expect-error
+		if (component && signal.owners.size > 0 && !signal.owners.has(component)) {
+			// eslint-disable-next-line no-console
+			console.warn(
+				'Mutating a value outside the component that created it is strongly discouraged. Consider passing values to child components with `bind:`, or use callback instead.'
```

Comment:
Do we have any more info we can share here? The stack trace tells you where the violation happened but it won't tell you who owns the state, which makes it harder to resolve the warning.

Reply:
added filenames, where possible